From 3f9f65a5a7df17794b5239acf0b6a4038c73b500 Mon Sep 17 00:00:00 2001
From: Isabelle Leonardi <isabellex.leonardi@intel.com>
Date: Mon, 10 Nov 2014 19:28:41 +0800
Subject: [PATCH 6/8] CDC_NCM: add FLAG_SEND_ZLP for intel modem

Padding with 0s in Tx NTBs is optional. If FLAG_SEND_ZLP flag is set,
no padding will be added, so it can improve the NCM throughput.

Change-Id: I3ed5d891188ffba4059366936281c0a40aa1de27
Signed-off-by: Isabelle Leonardi <isabellex.leonardi@intel.com>
Signed-off-by: Li Aixiong <aixiong.li@intel.com>
---
 drivers/net/usb/cdc_ncm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/usb/cdc_ncm.c b/drivers/net/usb/cdc_ncm.c
index 2ada082..307b543 100644
--- a/drivers/net/usb/cdc_ncm.c
+++ b/drivers/net/usb/cdc_ncm.c
@@ -1152,7 +1152,8 @@ static const struct driver_info wwan_noarp_info = {
 
 static const struct driver_info cdc_ncm_remote_mac_info = {
 	.description = "CDC NCM",
-	.flags = FLAG_POINTTOPOINT | FLAG_NO_SETINT | FLAG_MULTI_PACKET,
+	.flags = FLAG_POINTTOPOINT | FLAG_NO_SETINT | FLAG_MULTI_PACKET
+			| FLAG_SEND_ZLP,
 	.bind = cdc_ncm_remote_mac_bind,
 	.unbind = cdc_ncm_unbind,
 	.check_connect = cdc_ncm_check_connect,
-- 
1.9.1

